BGAnimations/ScreenSelectMusic overlay/PerPlayer/DensityGraph.lua

```diff
@@ -0,0 +1,137 @@
+local player = ...
+local pn = ToEnumShortString(player)
+local p = PlayerNumber:Reverse()[player]
+
+local height = 64
+local width = IsUsingWideScreen() and 286 or 276
+
+-- Max height at 280 BPM 16ths. Anything faster will just scale the graph. 
+local maxNps = 280/60 * 4
+
+local af = Def.ActorFrame{
+	InitCommand=function(self)
+		self:visible( GAMESTATE:IsHumanPlayer(player) )
+		self:xy(_screen.cx-182, _screen.cy+23)
+
+		if player == PLAYER_2 then
+			self:addy(height+24)
+		end
+
+		if IsUsingWideScreen() then
+			self:addx(-5)
+		end
+	end,
+	PlayerJoinedMessageCommand=function(self, params)
+		if params.Player == player then
+			self:visible(true)
+		end
+	end,
+	PlayerUnjoinedMessageCommand=function(self, params)
+		if params.Player == player then
+			self:visible(false)
+		end
+	end,
+
+	CurrentSongChangedMessageCommand=function(self) self:queuecommand("UpdateGraph") end,
+	CurrentCourseChangedMessageCommand=function(self) self:queuecommand("UpdateGraph") end,
+	CurrentStepsP1ChangedMessageCommand=function(self) self:queuecommand("UpdateGraph") end,
+	CurrentStepsP2ChangedMessageCommand=function(self) self:queuecommand("UpdateGraph") end,
+
+	UpdateGraphCommand=function(self)
+		if not GAMESTATE:IsCourseMode() and GAMESTATE:GetCurrentSong() then
+			self:GetChild("Breakdown"):visible(true)
+			self:GetChild("Histogram"):visible(true)
+			self:GetChild("NPS"):visible(false)
+		else
+			self:GetChild("Breakdown"):visible(false)
+			self:GetChild("Histogram"):visible(false)
+			self:GetChild("NPS"):settext("Peak NPS: ")
+			self:GetChild("NPS"):visible(false)
+		end
+	end,
+}
+
+-- Background quad for the density graph
+af[#af+1] = Def.Quad{
+	InitCommand=function(self)
+		self:diffuse(color("#1e282f")):zoomto(width, height)
+		if ThemePrefs.Get("RainbowMode") then
+			self:diffusealpha(0.9)
+		end
+	end
+}
+
+-- The Density Graph itself
+af[#af+1] = NPS_Histogram(player, width, height)..{
+	Name="Histogram",
```

Comment:
I think SM is not very happy about setting the name here:
```
/////////////////////////////////////////
WARNING: #1  ... in /Themes/_fallback/Scripts/02 ActorDef.lua:30
WARNING: #2  (tail call)
WARNING: #3  main in /Themes/Simply-Love-SM5/BGAnimations/ScreenSelectMusic overlay/PerPlayer/DensityGraph.lua:70
WARNING: #4  (tail call)
WARNING: #5  main in /Themes/Simply-Love-SM5/BGAnimations/ScreenSelectMusic overlay/PerPlayer/Over.lua:10
WARNING: #6  (tail call)
WARNING: #7  main in /Themes/Simply-Love-SM5/BGAnimations/ScreenSelectMusic overlay/default.lua:30
WARNING: 
WARNING: Overriding "Name": string with string
/////////////////////////////////////////
```
When scrolling through the songwheel the log gets spammed by those errors.

Reply:
Ah it's probably because the returned actor already has a name. I'll remove the name from here.

Reply:
Actually, I'm going to remove the name from the SL-Histogram.lua as it doesn't seem to be referenced anywhere else.